libpod/container_internal_linux.go

```diff
@@ -664,7 +664,7 @@ func (c *Container) makePlatformBindMounts() error {
 	// Make /etc/hostname
 	// This should never change, so no need to recreate if it exists
 	if _, ok := c.state.BindMounts["/etc/hostname"]; !ok {
-		hostnamePath, err := c.writeStringToRundir("hostname", c.Hostname())
+		hostnamePath, err := c.writeStringToRundir("hostname", c.Hostname() + "\n")
```

Comment:
Please do not do a string concat like this.

`fmt.Sprintf("%s\n", c.Hostname())` instead.

Reply:
no, it is better like this.
fmt.Sprintf is much slower and this really isn't a complex format where is using Sprintf is required or improves readability

Reply:
In general, I have seen "bad behaviors" resultant from just string concat. Specific case in point would be SQL injection attacks.

In terms of speed, this is negligible. If I were to be building 1000s of hostnames I would be concerned about the cumulative effect.

Additionally, and I apologize as I'm fairly new to the Go language, I have not gone into great depth of in terms of what the memory consumption would look like with this type of computation.

I'm fine with this either way as it reaches the goal.

Reply:
> In general, I have seen "bad behaviors" resultant from just string concat. Specific case in point would be SQL injection attacks.

Well Sprintf would not guard against that at all.

> In terms of speed, this is negligible. If I were to be building 1000s of hostnames I would be concerned about the cumulative effect.

Sure but if you do it everywhere it will get noticeable eventually. 
There is simply no benifit of fmt.Sprintf() unless it uses more complex specifiers so we should default to the faster variant.